pkgs/os-specific/linux/lenovo-legion/app.nix

```
@@ -0,0 +1,40 @@
+{ lib, fetchurl, xorg, libsForQt5, wrapQtAppsHook, python3 }:
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "lenovo-legion-app";
+  version = "main2023-04-02-16-53-51";
+
+  src = fetchurl {
+    url = "https://github.com/johnfanv2/LenovoLegionLinux/archive/refs/tags/${version}.tar.gz";
+    sha256 = "sha256-Ry7aSkFsmE7gTiYleZ08XUj7t9/MkyU4tpwfqMJA2GA=";
+  };
+
+  nativeBuildInputs = [ wrapQtAppsHook ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyqt5
+    argcomplete
+    pyyaml
+    xorg.libxcb
+    libsForQt5.qtbase
+  ];
+
+  preBuild = "cd python/legion_linux";
+
+  postInstall = ''
+    cp -r ./{legion.py,legion_cli.py,legion_gui.py} $out/lib/python3.10/site-packages
```

```suggestion
    cp -r ./{legion.py,legion_cli.py,legion_gui.py} $out/${python3.sitePackages}
```
this is important to change